src/utilities/command-utils.test.ts

```diff
@@ -0,0 +1,188 @@
+    // #region remove
+    // -----------------------------------------------------------------------------------------
+
+    describe("remove", () => {
+        test.each([undefined, null, "", " "])(
+            "when called with %p, it returns unmodified list of commands",
+            (name) => {
+                // Arrange
+                const command = seedCommandWithSubcommand(
+                    TestUtils.randomWord()
+                );
+                const expectedCommands = command.commands;
+
+                // Act
+                const result = CommandUtils.remove(
+                    command,
+                    (name as any) as string
+                );
+
+                // Assert
+                expect(result.commands).toBe(expectedCommands);
+            }
+        );
+
+        describe("given the command is not registered", () => {
+            test("when called with the command name, it returns unmodified list of commands", () => {
+                // Arrange
+                const name = TestUtils.randomWord();
+                const command = seedCommandWithSubcommand(`not-${name}`);
+                const expectedCommands = command.commands;
+
+                // Act
+                const result = CommandUtils.remove(command, name);
+
+                // Assert
+                expect(result.commands).toBe(expectedCommands);
+            });
+        });
+
+        describe("given a command is registered", () => {
+            test("when called with that name, it removes it from the list of commands", () => {
+                // Arrange
+                const name = TestUtils.randomWord();
+                const command = seedCommandWithSubcommand(name);
+                const unexpected = command.commands[0];
+
+                // Act
+                const result = CommandUtils.remove(command, name);
+
+                // Assert
+                expect(result).not.toBeNil();
+                expect(result.commands).not.toContain(unexpected);
+            });
+
+            test("when called with that name in any casing, it removes it from the list of commands", () => {
+                // Arrange
+                const name = TestUtils.randomWord();
+                const randomCaseName = TestUtils.randomCase(name);
+                const command = seedCommandWithSubcommand(name);
+                const unexpected = command.commands[0];
+
+                // Act
+                const result = CommandUtils.remove(command, randomCaseName);
+
+                // Assert
+                expect(result).not.toBeNil();
+                expect(result.commands).not.toContain(unexpected);
+            });
+        });
+    });
+
+    // #endregion remove
+
+    // -----------------------------------------------------------------------------------------
+    // #region sort
+    // -----------------------------------------------------------------------------------------
+
+    describe("sort", () => {});
```

Comment:
Should there be a github issue created and linked here to backfill?

Reply:
Good call 👍  Just hadn't gotten to it yet. I'll add issues and link them